### PR TITLE
Provide confidential values as secret

### LINKED ISSUE #70: Feature: Externalize secrets
## Feature Description

Currently, I have to configure confidential values in the 'values.yaml'. So this is also true for local values.yaml file for which I want to version configuration settings but not the secret ones.

## Proposed Solution

Provide a secret sample including demo setting (from the current values.yaml) and make the name configurable. Then, we can reference to secrets in the env section of each deployment. 

In case of overriding, one would have to create a custom secret file within a dedicated configure and configure that secret name. The secret can be safely ignored by version management while all other configuration can be put under version control.

### PR DESCRIPTION
**Pull request by ridoo**
## Description

Separates confidential values (currently the superuser credentials) from `values.yaml` into a Kubernetes Secret. 
Provides a `templates/demo-secret.yaml` by default. However, can be overridden by a custom secret which can be referenced by name in the `values.yaml`.

@mwallschlaeger please have a look, if you think this is a path we may want to follow. If you are ok, I will continue refactoring the remaining credentials.

## Type of Change

Please select the relevant option:

- [ ] Bug fix
- [x] New feature
- [x] Documentation update
- [ ] Refactoring
- [ ] Other (please describe)

## Related Issue

If there is an existing issue related to this pull request, please reference it here.

closes #70 

## Checklist

Please ensure that your pull request meets the following requirements:

- The pull request is limited to one type (docs, feature, bug fix, etc.)
- The pull request is as small as possible. Consider opening multiple pull requests instead of one large one.
- The feature or bug fix has been discussed and documented in an issue beforehand.

## Additional Notes

Any additional information or context regarding the pull request can be provided here.

Thank you for creating this pull request

### CODE REVIEW

File: charts/geonode/values.yaml
Comment (mwallschlaeger) [inline charts/geonode/values.yaml:1]: I agree on storing sensitive data in secrets makes sense. But I don't understand why you want to deploy them  separately. In my opinion this makes the deployment more complex. Could we keep the definition inside the values.yaml but store them in kubernetes secrets? Generally securing sensitive data in the values.yaml on the deployment server  should be part of the CI tool, like argocd or flux. Those tools provide mechanisms to secure secrets in values.yaml like `sops` or `vault`.

**Review by mwallschlaeger — CHANGES_REQUESTED** [review]
see my two remarks on this. else i'm fine with the general idea on this now.

File: charts/geonode/templates/geoserver/geoserver-deploy.yaml [hunk 1]
```diff
@@ -65,6 +65,8 @@ spec:
         envFrom:
           - configMapRef:
               name:  {{ include "geoserver_pod_name" . }}-env
+          - secretRef:
+              name: {{ .Values.geoserver.secret.existingSecretName | "geoserver-secret" | quote }}
```
Comment (mwallschlaeger) [inline charts/geonode/templates/geoserver/geoserver-deploy.yaml:69]: See comments on geonode-deploy.yaml herefor

**Review by mwallschlaeger — APPROVED** [review]

File: charts/geonode/templates/geonode/geonode-deploy.yaml [hunk 2]
```diff
@@ -89,6 +89,10 @@ spec:
         envFrom:
           - configMapRef:
               name: {{ include "geonode_pod_name" . }}-env
+          - secretRef:
+            name: {{ .Values.geonode.secret.existingSecretName | "geonode-secret" | quote }}
+          - secretRef:
+            name: {{ .Values.geoserver.secret.existingSecretName | "geoserver-secret" | quote }}
```
[the remark this answers is not in the document]
Comment (ridoo) [reply charts/geonode/templates/geonode/geonode-deploy.yaml]: Yes, I saw it already and provided a solutions using `default`

File: charts/geonode/templates/geonode/geonode-secret.yaml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+kind: Secret
+metadata:
+  name: geonode-secret
+  namespace: {{ .Release.Namespace }}
+type: Opaque
+data:
+  # superuser credentials
+  ADMIN_USERNAME: {{ .Values.geonode.secret.content.superUser.username | b64enc }}
+  ADMIN_PASSWORD: {{ .Values.geonode.secret.content.superUser.password | b64enc }}
+  ADMIN_EMAIL: {{ .Values.geonode.secret.content.superUser.email | b64enc }}
```
[the remark this answers is not in the document]
Comment (ridoo) [reply charts/geonode/templates/geonode/geonode-secret.yaml]: fixed it

File: charts/geonode/templates/postgres/postgres-external-geonode-secrets.yaml [hunk 4]
```diff
@@ -1,10 +1,11 @@
-{{ if .Values.postgres.external_postgres.enabled }}
+{{ if and ( .Values.postgres.external_postgres.enabled (empty .Values.postgres.external_postgres.secret.existingSecretName )) }}
```
Comment (ridoo) [inline charts/geonode/templates/postgres/postgres-external-geonode-secrets.yaml:1]: This does not work -- will provide a fix and unblock

Comment (ridoo) [reply charts/geonode/templates/postgres/postgres-external-geonode-secrets.yaml]: https://repeatit.io/ FTW

File: docs/provide-custom-secret.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+1. Override the `secret.content` to set project specific secrets
+1. Create custom Kubernetes Secrets and configure `secret.name` accordingly
+
+> :bulb: **Note:**
+>
+> Make sure to not expose your secrets, e.g. via Git! 
+> Consider to pass secrets from a CD pipeline via masked environment settings.
+> By referencing a custom Secret (leaving `secret.content` empty), you would have to create and maintain a Secret by yourself.
+
```
Comment (mwallschlaeger) [inline docs/provide-custom-secret.md:16]: this isn't up-to-date with the latest changes to this pull request anymore

**Review by mwallschlaeger — CHANGES_REQUESTED** [review]

**Review by mwallschlaeger — APPROVED** [review]

**Pull request merged**